File: app/src/main/java/com/example/installer/view/FilterTabItemView.java
package com.example.installer.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.RelativeLayout;

import com.example.installer.R;
import com.example.installer.databinding.ViewFilterTabItemBinding;


/**
 * Implementation of App Widget functionality.
 */
public class FilterTabItemView extends RelativeLayout {
    private ViewFilterTabItemBinding binding;
    private boolean isHighlight;

    public FilterTabItemView(Context context) {
        super(context);
        initView();
    }

    public FilterTabItemView(Context context, AttributeSet attrs) {
        super(context, attrs);
        initView();
    }

    public FilterTabItemView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initView();
    }

    private void initView() {
        binding = ViewFilterTabItemBinding.inflate(LayoutInflater.from(this.getContext()), this,true);
    }

    public void setTitle(String title) {
        binding.tvFilterTitle.setText(title);
    }

    public boolean isHighlight() {
        return isHighlight;
    }

    public void setHighlight(boolean highlight) {
        isHighlight = highlight;
        if (isHighlight) {
            binding.tvFilterTitle.setTextColor(getResources().getColor(R.color.titleTextSelectedColor));
            binding.tvFilterTitle.getCompoundDrawables()[2].setLevel(1);
        } else {
            binding.tvFilterTitle.setTextColor(getResources().getColor(R.color.titleTextDefaultColor));
            binding.tvFilterTitle.getCompoundDrawables()[2].setLevel(0);
        }
    }
}

File: app/src/main/java/com/example/installer/view/FilterTabItemView.kt
package com.example.installer.view

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.RelativeLayout
import androidx.core.content.ContextCompat
import com.example.installer.R
import com.example.installer.databinding.ViewFilterTabItemBinding

/**
@author zhangtao
@date   2020/10/28

 **/
class FilterTabItemView : RelativeLayout {
    private lateinit var binding: ViewFilterTabItemBinding
    private var isHighlight = false

    constructor(context: Context) : super(context) {
        initView(context)
    }

    constructor(context: Context, attributes: AttributeSet) : super(context, attributes) {
        initView(context)
    }

    constructor(context: Context, attributes: AttributeSet, defStyleAttr: Int) : super(
        context,
        attributes,
        defStyleAttr
    ) {
        initView(context)
    }

    private fun initView(context: Context) {
        binding = ViewFilterTabItemBinding.inflate(LayoutInflater.from(context), this, true)
    }

    fun setTitle(title: String?) {
        binding.tvFilterTitle.text = title
    }

    fun setHighLight(highLight: Boolean) {
        this.isHighlight = highLight
        if (isHighlight) {
            binding.tvFilterTitle.setTextColor(
                ContextCompat.getColor(
                    context,
                    R.color.titleTextSelectedColor
                )
            )
            binding.tvFilterTitle.compoundDrawables[2].level = 1
        } else {
            binding.tvFilterTitle.setTextColor(
                ContextCompat.getColor(
                    context,
                    R.color.titleTextDefaultColor
                )
            )
            binding.tvFilterTitle.compoundDrawables[2].level = 0
        }
    }
}